Mathlib/Data/ZMod/Module.lean

```diff
@@ -42,3 +42,36 @@ theorem toZModLinearMap_injective: Function.Injective <| toZModLinearMap n (M :=
 theorem coe_toZModLinearMap (f : M →+ M₁) : ⇑(f.toZModLinearMap n) = f := rfl
 
 end AddMonoidHom
+
+namespace AddSubgroup
+
+/-- Reinterpret an additive subgroup of a `ℤ/nℤ`-module as a `ℤ/nℤ`-submodule.
+
+See also: `AddSubgroup.toIntSubmodule`, `AddSubmonoid.toNatSubmodule`. -/
+def toZModSubmodule : AddSubgroup M ≃o Submodule (ZMod n) M where
+  toFun S := { S with smul_mem' := fun c _ h ↦ ZMod.smul_mem (K := S) h c }
+  invFun := Submodule.toAddSubgroup
+  left_inv _ := rfl
+  right_inv _ := rfl
+  map_rel_iff' := Iff.rfl
+
+@[simp]
```

Comment:
I had the idea to copy these simp lemmas from `toIntSubmodule`, but is there a way to automate these?